File: EventFlowAPI/EventFlowAPI.DB/Entities/Ticket.cs
using EventFlowAPI.DB.Entities.Abstract;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EventFlowAPI.DB.Entities
{
    public class Ticket : BaseEntity, ISoftDeleteable, IUpdateableEntity
    {

        [Range(0.00, 999.99),
         Column(TypeName = "NUMERIC(5,2)")]
        public decimal Price { get; set; }
        public bool IsDeleted { get; set; } = false;
        public DateTime? DeleteDate { get; set; }
        public bool IsUpdated { get; set; } = false;
        public DateTime? UpdateDate { get; set; }
        public int TicketTypeId { get; set; }
        public int EventId { get; set; }
        public int? FestivalId { get; set; }
        public Event Event { get; set; } = default!;
        public Festival? Festival { get; set; }
        public TicketType TicketType { get; set; } = default!;
        public ICollection<Reservation> Reservations { get; set; } = [];
    }
}

File: EventFlowClient/src/models/response_models/Ticket.ts
import { IModel } from "../abstract/IModel";
import { EventEntity } from "./EventEntity";
import { Festival } from "./Festival";
import { TicketType } from "./TicketType";

export type Ticket = IModel & {
  price: number;
  isFestival: boolean;
  event?: EventEntity;
  festival?: Festival;
  ticketType: TicketType;
};
